pyquil/operator_estimation.py

```diff
@@ -814,11 +816,101 @@ def serializable(self):
+
+    :param qc: a quantum computer object on which to run the programs necessary to calibrate each
+        result.
+    :param expt_results: a list of results, each of which will be separately calibrated.
+    :param n_shots: the number of shots to run for each eigenvector
+    :param symm_type: the type of symmetrization
+
+        * -1 -- exhaustive symmetrization uses every possible combination of flips; this option
+            is the default since it ensures proper calibration, but is exponential in the
+            weight of each observable.
+        * 0 -- no symmetrization
+        * 1 -- symmetrization using an OA with strength 1
+        * 2 -- symmetrization using an OA with strength 2
+        * 3 -- symmetrization using an OA with strength 3
+
+        TODO: accomodate calibration for weight > symmetrization strength (symm_type)
+        Currently, the symmetrization type must be at least the maximum weight of any observable
+        estimated and also match the symmetrization type used to estimate the observables of the
+        input ExperimentResults.
+    :param noisy_program: an optional program from which to inherit a noise model; only relevant
+        for running on a QVM
+    :param active_reset: whether or not to begin the program by actively resetting. If true,
+        execution of each of the returned programs in a loop on the QPU will generally be faster.
+    :param show_progress_bar: displays a progress bar via tqdm if true.
+    :return: a copy of the input results with updated estimates and calibration results.
+    """
+    # get unique observables that will need to be calibrated
+    observables = {copy(res.setting.out_operator) for res in expt_results}
+
+    calibrations = {}
+    for obs in tqdm(observables, disable=not show_progress_bar):
+        prog = get_calibration_program(obs, noisy_program, active_reset)
+        meas_qs = obs.get_qubits()
+        results = qc.run_symmetrized_readout(prog, n_shots, symm_type, meas_qs)
+
+        # Obtain statistics from result of experiment
+        # TODO: we have to fabricate an ExperimentSetting to pass to _stats_from_measurements
+        #  even though it only needs the observable.
+        setting = ExperimentSetting(zeros_state(meas_qs), obs)
+        obs_mean, obs_var = _stats_from_measurements(results,
+                                                     {q: idx for idx, q in enumerate(meas_qs)},
+                                                     setting, len(results))
+        calibrations[obs.operations_as_set()] = (obs_mean, obs_var, len(results))
+
+    for expt_result in expt_results:
+        # TODO: allow weight > symm_type
+        if -1 < symm_type < len(expt_result.setting.out_operator.get_qubits()):
+            warnings.warn(f'Calibration of observable {expt_result.setting.out_operator} '
+                          f'currently not supported since it acts on more qubits than the '
+                          f'symm_type {symm_type}.')
```

Comment:
would it make sense to skip the `expt_result` in this case, or better to proceed anyway?

Reply:
I'm not sure. I went with a warning because I'm not sure how to interpret such a calibration, but maybe it isn't a problem depending on the error model.